src/libcore/tests/lib.rs

```diff
@@ -32,8 +32,8 @@
 #![feature(slice_partition_dedup)]
 #![feature(copy_within)]
 #![feature(int_error_matching)]
+#![deny(rust_2018_idioms)]
 
-extern crate core;
 extern crate test;
```

Comment:
Any idea why you didn't also have to remove this line? It causes a build failure when building the tests "from the outside" at https://travis-ci.org/RalfJung/miri-test-libstd/builds/523334973.

Reply:
I'm not really sure. Do you maybe have to pass `--test` in order for the crate to be used?

Reply:
Hm, no... removing it actually makes it fail in `x.py`.  But somehow it is considered redundant when compiled in a less magic environment?

Reply:
I think adding `#[allow(unused_extern_crates)]` before `extern crate test;` should fix it (like [here](https://github.com/rust-lang/rust/blob/d6f513ec7d1e83c8689f94fb48686dd11f1d1c80/src/librustc/lib.rs#L94)), but I won't be able to actually try it out until later today.

Reply:
This should also work: https://github.com/rust-lang/rust/pull/60201